File: frontend/src/types/schedule.ts
export type ScheduleStatus = 'pending' | 'published' | 'failed'

export interface Schedule {
  id: string
  post_id: string
  post_title: string | null
  platform: string
  platform_account_id: string
  platform_account_label: string | null
  scheduled_at: string
  timezone: string
  status: ScheduleStatus
  retry_count: number
  last_error: string | null
  published_at: string | null
  created_at: string
}

export interface ScheduleListResponse {
  items: Schedule[]
  total: number
}

File: backend/app/schemas/schedule.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class ScheduleCreate(BaseModel):
    platform_account_id: uuid.UUID
    scheduled_at: datetime
    timezone: str = "UTC"


class ScheduleReschedule(BaseModel):
    scheduled_at: datetime
    timezone: str = "UTC"
    platform_account_id: uuid.UUID | None = None


class ScheduleOut(BaseModel):
    id: uuid.UUID
    post_id: uuid.UUID
    post_title: str | None
    platform: str
    platform_account_id: uuid.UUID
    platform_account_label: str | None
    scheduled_at: datetime
    timezone: str
    status: str
    retry_count: int
    last_error: str | None
    published_at: datetime | None
    created_at: datetime


class ScheduleListResponse(BaseModel):
    items: list[ScheduleOut]
    total: int
